### DISCUSSION TITLE
Is it possible to generate a set of patients with encounter with specific year? If so, how? Thanks

### DISCUSSION THREAD

**Discussion by mduna** [synthetichealth/synthea#1017] [Q&A]

**Comment by jawalonoski** [comment]
No.

Generally, that is not how Synthea works. It is a simulation of events, not a declarative shopping cart.

That being said, you could theoretically do what you are asking for using a [Keep Patients Module](https://github.com/synthetichealth/synthea/wiki/Keep-Patients-Module), but the logic would be twisted and I'm not sure how to write it... because you can't currently directly reference encounters or prior states in other modules using the built-in conditional logic.

A work around is to generate more patients than you need, and then post-process the data and filter the list of patients down to those that have the specific encounter type to the specific year of interest.

**Reply by mduna** [reply]
Thank you for quick response! Could Synthea limit the years for all patient's clinical data? for example only for two years?

**Reply by aiya6502** [reply]
May be you can edit the CSV fie and transform the years to your desired one.

**Comment by hadleynet** [comment]
Mostly, try these two command line switches: `--exporter.years_of_history=2 --generate.only_alive_patients=true` this should help reduce the amount of data.

**Comment by jawalonoski** [comment]
Another idea:

```json
{
  "name": "example",
  "remarks": [
    "Example of a module with an encounter that can only occur in 2022."
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "direct_transition": "Only 2022"
    },
    "Terminal": {
      "type": "Terminal"
    },
    "Only 2022": {
      "type": "Guard",
      "allow": {
        "condition_type": "Date",
        "operator": "==",
        "year": 2022,
        "value": 0
      },
      "direct_transition": "Specific Year Encounter"
    },
    "Specific Year Encounter": {
      "type": "Encounter",
      "encounter_class": "ambulatory",
      "reason": "",
      "telemedicine_possibility": "none",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": "1234",
          "display": "SNOMED Code"
        }
      ],
      "direct_transition": "End Encounter"
    },
    "End Encounter": {
      "type": "EncounterEnd",
      "direct_transition": "Terminal"
    }
  },
  "gmf_version": 2
}
```

**Reply by mduna** [reply]
Thanks, it works for me to limit the Encounter period which I want!
Can Synthea create Specimen Resource? If so, could you please give me some direction? Thanks.

**Reply by jawalonoski** [reply]
Synthea does not currently support the FHIR `Specimen` resource.

**Reply by mduna** [reply]
Thanks.

**Reply by mduna** [reply]
Is it possible to make multiple medication administrations for a given medication? For example, a vancomycin IV medication once a day for 5 days. Synthea will generate five MedicationRequest resources for five days, one resource per day.  If it is possible, could please show me an example? Thanks.

**Reply by jawalonoski** [reply]
It would be nice if you could accept one of the answers to your previous question, if any of them met your need, and then ask your new medication administration question as a separate discussion.

**Reply by mduna** [reply]
Yes, your answer to my previous question is very helpful. I will ask the medication administration question in a separate discussion. Thanks.

**Comment by PAastha** [comment]
i am not able to generate encounter specific to a year through the json file mentioend above.
